guides/04_chatbots/01_creating-a-chatbot-fast.md

```diff
@@ -0,0 +1,254 @@
+
+## Using your chatbot via an API
+
+Once you've built your Gradio chatbot and are hosting it on [Hugging Face Spaces](https://hf.space) or somewhere else, then you can query it with a simple API at the `/chat` endpoint. The endpoint just expects the user's message, and will return the response, internally keeping track of the messages sent so far.
+
+[](!https://github.com/gradio-app/gradio/assets/1778297/7b10d6db-6476-4e2e-bebd-ecda802c3b8f)
+
+To use the endpoint, you should use either the [Gradio Python Client](/guides/getting-started-with-the-python-client) or the [Gradio JS client](/guides/getting-started-with-the-js-client).
+
+## A `langchain` example
+
+Now, let's actually use the `gr.ChatInterface` with some real large language models. We'll start by using `langchain` on top of `openai` to build a general-purpose streaming chatbot application in 19 lines of code. You'll need to have an OpenAI key for this example (keep reading for the free, open-source equivalent!)
+
+```py
+from langchain.chat_models import ChatOpenAI
+from langchain.schema import AIMessage, HumanMessage
+import openai
+import gradio as gr
+
+openai.api_key = os.getenv("OPENAI_API_KEY")
+
+llm = ChatOpenAI(temperature=1.0, model='gpt-3.5-turbo-0613')
+
+def predict(message, history):
+    history_langchain_format = []
+    for human, ai in history:
+        history_langchain_format.append(HumanMessage(content=human))
+        history_langchain_format.append(AIMessage(content=ai))
+    history_langchain_format.append(HumanMessage(content=message))
+    gpt_response = chat(history_langchain_format)
+    return gpt_response.content
+
+gr.ChatInterface(predict).launch() 
+```
+
+## A streaming example using `openai`
+
+Of course, we could also use the `openai` library directy. Here a similar example, but this time with streaming results as well:
+
+
+```py
+import openai
+import gradio as gr
+
+openai.api_key = os.getenv("OPENAI_API_KEY")
+
+def predict(message, history):
+    history_openai_format = []
+    for human, assistant in history:
+        history_openai_format.append({"role": "user", "content": human })
+        history_openai_format.append({"role": "assistant", "content":assistant})
+    history_openai_format.append({"role": "user", "content": message})
+
+    response = openai.ChatCompletion.create(
+        model='gpt-3.5-turbo',
+        messages= history_openai_format,         
+        temperature=1.0,
+        stream=True
+    )
+    
+    partial_message = ""
+    for chunk in response:
+        if len(chunk['choices'][0]['delta']) != 0:
+            partial_message = partial_message + chunk['choices'][0]['delta']['content']
+            yield partial_message 
+
+gr.ChatInterface(predict).queue().launch() 
+```
+
+## Examples using open-source LLMs with Hugging Face!
+
+Of course, why use a closed-source model when you can use an open-source one instead? Here's the equivalent example using Together's RedePajama model.
+
+```py
```

Comment:
Should we deploy as a space so that we can embed it in the docs and have users try it out? WDYT @yvrjsharma ?